src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -20,18 +35,169 @@ import {CDK_MENU, Menu} from './menu-interface';
   selector: '[cdkMenuBar]',
   exportAs: 'cdkMenuBar',
   host: {
+    '(keydown)': '_handleKeyEvent($event)',
+    '(focus)': 'focusFirstItem()',
     'role': 'menubar',
+    'tabindex': '0',
     '[attr.aria-orientation]': 'orientation',
   },
   providers: [
     {provide: CdkMenuGroup, useExisting: CdkMenuBar},
     {provide: CDK_MENU, useExisting: CdkMenuBar},
+    {provide: MenuStack, useClass: MenuStack},
   ],
 })
-export class CdkMenuBar extends CdkMenuGroup implements Menu {
+export class CdkMenuBar extends CdkMenuGroup implements Menu, AfterContentInit {
   /**
    * Sets the aria-orientation attribute and determines where menus will be opened.
    * Does not affect styling/layout.
    */
   @Input('cdkMenuBarOrientation') orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Handles keyboard events for the MenuBar. */
+  private _keyManager: FocusKeyManager<CdkMenuItem>;
+
+  /** All child MenuItem elements nested in this MenuBar. */
+  @ContentChildren(CdkMenuItem, {descendants: true})
+  private readonly _allItems: QueryList<CdkMenuItem>;
+
+  constructor(private readonly _dir: Directionality, readonly _menuStack: MenuStack) {
+    super();
+  }
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    this._setKeyManager();
+    this._subscribeToMenuStack();
+  }
+
+  /** Place focus on the first MenuItem in the menu and set the focus origin. */
+  focusFirstItem(focusOrigin: FocusOrigin = 'program') {
+    this._keyManager.setFocusOrigin(focusOrigin);
+    this._keyManager.setFirstItemActive();
+  }
+
+  /** Place focus on the last MenuItem in the menu and set the focus origin. */
+  focusLastItem(focusOrigin: FocusOrigin = 'program') {
+    this._keyManager.setFocusOrigin(focusOrigin);
+    this._keyManager.setLastItemActive();
+  }
+
+  /**
+   * Handle keyboard events, specifically changing the focused element and/or toggling the active
+   * items menu.
+   * @param event the KeyboardEvent to handle.
+   */
+  _handleKeyEvent(event: KeyboardEvent) {
+    switch (event.keyCode) {
+      case UP_ARROW:
+      case DOWN_ARROW:
+      case LEFT_ARROW:
+      case RIGHT_ARROW:
+        const horizontalArrows = event.keyCode === LEFT_ARROW || event.keyCode === RIGHT_ARROW;
+        // For a horizontal menu if the left/right keys were clicked, or a vertical menu if the
+        // up/down keys were clicked: if the current menu is open, close it then focus and open the
+        // next  menu.
+        if (
+          (this._isHorizontal() && horizontalArrows) ||
+          (!this._isHorizontal() && !horizontalArrows)
+        ) {
+          event.preventDefault();
+
+          const prevIsOpen = this._keyManager.activeItem?.isMenuOpen();
```

Comment:
It looks like you're repeating `this._keyManager.activeItem` in almost all of the switch cases. Maybe move it out into a variable at the top?